doctor/routetable/routetable_bsd.go

```diff
@@ -0,0 +1,256 @@
+			}
+			fmt.Fprintf(&sb, "%02x", x)
+		}
+		re.GatewayAddr = sb.String()
+	}
+}
+
+func populateDestination(re *routeEntry, ifs map[int]interfaces.Interface, rm *route.RouteMessage) {
+	// Default destination is "invalid" if we don't parse further
+	re.Dst = "invalid"
+
+	dst := rm.Addrs[unix.RTAX_DST]
+	if dst == nil {
+		return
+	}
+
+	ip := ipFromRMAddr(ifs, dst)
+	if !ip.IsValid() {
+		return
+	}
+
+	if ip.Is4() {
+		re.Family = unix.AF_INET
+	} else {
+		re.Family = unix.AF_INET6
+	}
+	re.Dst = ip.String() // default if nothing more specific
+
+	// If the RTF_HOST flag is set, then this is a host route and there's
+	// no netmask in this RouteMessage.
+	if rm.Flags&unix.RTF_HOST != 0 {
+		return
+	}
+
+	// As above if there's no netmask in the list of addrs
+	if len(rm.Addrs) < unix.RTAX_NETMASK || rm.Addrs[unix.RTAX_NETMASK] == nil {
+		return
+	}
+
+	nm := ipFromRMAddr(ifs, rm.Addrs[unix.RTAX_NETMASK])
+	if !ip.IsValid() {
+		return
+	}
+
+	// Count the number of bits in the netmask IP and use that to make our prefix.
+	ones, _ /* bits */ := net.IPMask(nm.AsSlice()).Size()
+	if ip.IsUnspecified() && ones == 0 {
+		re.Dst = "default"
+		return
+	}
+
+	// Print this ourselves instead of using netip.Prefix so that we don't
+	// lose the zone (since netip.Prefix strips that).
+	//
+	// NOTE(andrew): this doesn't print the same values as the 'netstat' tool
+	// for some addresses on macOS, and I have no idea why. Specifically,
+	// 'netstat -rn' will show something like:
+	//    ff00::/8   ::1      UmCI     lo0
+	//
+	// But we will get:
+	//    destination=ff00::/40 [...]
+	//
+	// The netmask that we get back from FetchRIB has 32 more bits in it
+	// than netstat prints, but only for multicast routes.
+	//
+	// For consistency's sake, we're going to do the same here so that we
+	// get the same values as netstat returns.
+	if runtime.GOOS == "darwin" && ip.Is6() && ip.IsMulticast() && ones > 32 {
+		ones -= 32
+	}
+	re.Dst = ip.String() + "/" + strconv.Itoa(ones)
+}
+
+func routeEntryFromMsg(ifsByIdx map[int]interfaces.Interface, msg route.Message) (routeEntry, bool) {
+	rm, ok := msg.(*route.RouteMessage)
+	if !ok {
+		return routeEntry{}, false
+	}
+
+	// Ignore things that we don't understand
+	if rm.Version < 3 || rm.Version > 5 {
+		return routeEntry{}, false
+	}
+	if rm.Type != rmExpectedType {
+		return routeEntry{}, false
+	}
+	if len(rm.Addrs) < unix.RTAX_GATEWAY {
+		return routeEntry{}, false
+	}
+
+	if rm.Flags&skipFlags != 0 {
+		return routeEntry{}, false
+	}
+
+	re := routeEntry{
+		rawFlags: rm.Flags,
+	}
+	for fv, fs := range flags {
+		if rm.Flags&fv == fv {
+			re.Flags = append(re.Flags, fs)
+		}
+	}
+	sort.Strings(re.Flags)
+	populateDestination(&re, ifsByIdx, rm)
+	if unix.RTAX_GATEWAY < len(rm.Addrs) {
+		populateGateway(&re, ifsByIdx, rm.Addrs[unix.RTAX_GATEWAY])
+	}
+
+	if outif, ok := ifsByIdx[rm.Index]; ok {
+		re.OutputIf = outif.Name
+	}
+	return re, true
+}
+
+func getRouteTable() ([]routeEntry, error) {
```

Comment:
Yeah, I was struggling to find a good cross-platform abstraction for "give me a route table" since the concept of a route table isn't particularly cross-platform.

Reply:
It's mostly cross platform, though. There are some per-OS special fields but we could shove them into a 'Sys any` field/method like https://pkg.go.dev/io/fs#FileInfo and its Sys.